feat(STONEINTG-765): remove outdated info in surface-tests doc (#244)
redhat-appstudio/docs.appstudio.io: docs/modules/ROOT/pages/how-to-guides/testing_applications/surface-level_tests.adoc

This document covers the surface-level tests (formerly known as "sanity tests") that {ProductName} runs as part of its component build pipeline. These surface-level tests automatically check all application images to ensure that they're up-to-date, correctly formatted, and protected from security vulnerabilities.

== Surface-level tests
The {ProductName} component build pipeline supports several types of tests, including surface-level tests. The surface-level tests used in {ProductName} are run in the form of https://tekton.dev/docs/pipelines/tasks/#overview[Tekton tasks]. The utility used for validating container information is https://www.conftest.dev/[Conftest]. The following tables show the currently implemented surface-level tests:

.Deprecated image checks
|===
|Test name |Description |Failure message

> |image_repository_deprecated |Deprecated images are no longer maintained and will accumulate security issues without releasing a fixed version | The container image should not be built from a repository which is marked as 'Deprecated' in COMET

Reviewer: ```suggestion
|image_repository_deprecated |Deprecated images are no longer maintained and will accumulate security issues without releasing a fixed version. | The container image should not be built from a repository which is marked as 'Deprecated' in COMET.
```

> |image_repository_deprecated |Deprecated images are no longer maintained and will accumulate security issues without releasing a fixed version | The container image should not be built from a repository which is marked as 'Deprecated' in COMET

Reviewer: ```suggestion
|image_repository_deprecated |Deprecated images are no longer maintained, leading to unresolved security vulnerabilities. | The container image must not be built from a repository  marked as 'Deprecated' in COMET
```

Author: Ack, updated.

|===

.Unsigned RPM check
|===
|Test name |Description |Failure message

> |image_unsigned_rpms |Providing packages signed with the secure Red Hat signing server indicates that the package was subject to all appropriate policies and procedures |All RPMs found on the image must be signed. Found following unsigned rpms(nvra):

Reviewer: ```suggestion
|image_unsigned_rpms |Packages signed with Red Hat's secure signing server adheres to stringent policies and procedures. |All RPMs in the image must be signed. Found following unsigned rpms(nvra):
```

Author: Ack, updated.
